test/micro/org/openjdk/bench/java/lang/invoke/LazyStaticColdStart.java

```diff
@@ -0,0 +1,155 @@
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.lang.constant.ClassDesc;
+import java.lang.constant.MethodTypeDesc;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.constant.ConstantDescs.*;
+import static jdk.internal.classfile.Classfile.ACC_STATIC;
+
+/**
+ * A benchmark ensuring that var and method handle lazy initialization are not
+ * too slow compared to eager initialization.
+ */
+@BenchmarkMode(Mode.SingleShotTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Thread)
+@Fork(value = 10, warmups = 5, jvmArgsAppend = {
+        "--add-exports", "java.base/jdk.internal.classfile=ALL-UNNAMED",
+        "--add-exports", "java.base/jdk.internal.classfile.attribute=ALL-UNNAMED",
+        "--add-exports", "java.base/jdk.internal.classfile.constantpool=ALL-UNNAMED",
+        "--add-exports", "java.base/jdk.internal.classfile.instruction=ALL-UNNAMED",
+        "--add-exports", "java.base/jdk.internal.classfile.components=ALL-UNNAMED"
+})
+public class LazyStaticColdStart {
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    private Class<?> targetClass;
+
+    /**
+     * Ensures non-initialized targetClass is used and initializes the lazy/non-lazy handles
+     * to prevent further creation costs.
+     */
+    @Setup(Level.Iteration)
+    public void setup() throws Throwable {
+        class Holder {
+            static final ClassDesc describedClass = LazyStaticColdStart.class.describeConstable().orElseThrow().nested("Data");
+            static final ClassDesc CD_ThreadLocalRandom = ThreadLocalRandom.class.describeConstable().orElseThrow();
+            static final ClassDesc CD_Blackhole = Blackhole.class.describeConstable().orElseThrow();
+            static final MethodTypeDesc MTD_void_long = MethodTypeDesc.of(CD_void, CD_long);
+            static final MethodTypeDesc MTD_ThreadLocalRandom = MethodTypeDesc.of(CD_ThreadLocalRandom);
+            static final MethodTypeDesc MTD_long = MethodTypeDesc.of(CD_long);
+            static final byte[] classBytes = Classfile.build(describedClass, clb -> {
+                clb.withField("v", CD_long, ACC_STATIC);
+                clb.withMethodBody(CLASS_INIT_NAME, MTD_void, ACC_STATIC, cob -> {
+                    cob.constantInstruction(100L);
+                    cob.invokestatic(CD_Blackhole, "consumeCPU", MTD_void_long);
+                    cob.invokestatic(CD_ThreadLocalRandom, "current", MTD_ThreadLocalRandom);
+                    cob.invokevirtual(CD_ThreadLocalRandom, "nextLong", MTD_long);
+                    cob.putstatic(describedClass, "v", CD_long);
+                    cob.return_();
+                });
+            });
+
+            static {
```

Comment:
Is this static block required?

Reply:
I use this block to remove the overhead around MethodHandle and VarHandle infrastructure initialization (such as LF spinning) that is not related to any particular class; so that I measure only the impact of encountering a new class.

Reply:
Ah, implied by "... and initializes the lazy/non-lazy handles to prevent further creation costs" Could you also add a small comment above this block?

Reply:
Done. Also commented on each line about what each call initializes.